src/software/embedded/robot_diagnostics_cli/robot_diagnostics_cli.py

```diff
@@ -0,0 +1,396 @@
+import subprocess
+import logging
+import typer as Typer
+from rich import print
+from rich.live import Live
+from rich.table import Table
+from rich.console import Console
+from rich.logging import RichHandler
+from software.py_constants import *
+from typer_shell import make_typer_shell
+from functools import wraps
+from typing import List, Optional
+from typing_extensions import Annotated
+from embedded_communication import EmbeddedCommunication
+from proto.import_all_protos import *
+from software.embedded.constants.py_constants import (
+    DEFAULT_PRIMITIVE_DURATION,
+    ROBOT_MAX_ANG_SPEED_RAD_PER_S,
+    ROBOT_MAX_SPEED_M_PER_S,
+    MAX_FORCE_DRIBBLER_SPEED_RPM,
+)
+
+
+class RobotDiagnosticsCLI:
+    """Onboard lightweight Diagnostics CLI interface running on the robots. UI and Main class"""
+
+    def __init__(self, embedded_communication: EmbeddedCommunication) -> None:
+        """Setup constructor for the Shell CLI
+        :param embedded_communication: Communication object with open connection to robots for sending protos
+        """
+        intro = """[bold yellow]Welcome to the RobotDiagnostics CLI [/bold yellow]
+        
+        Type [bold blue]help[/bold blue] for a list of commands. 
+        Use [bold blue]--help[/bold blue] in commands for extra info. 
+        If you ever have questions feel free to reach out!"""
+
+        self.app = make_typer_shell(
+            prompt="⚡ ", launch=lambda x: print(intro), intro="\n"
+        )
+        self.app.command(short_help="Moves the robot")(self.move)
+        self.app.command(short_help="Moves specific wheels of the robot")(
+            self.move_wheel
+        )
+        self.app.command(short_help="Rotates the robot")(self.rotate)
+        self.app.command(short_help="Spins the dribbler")(self.dribble)
+        self.app.command(short_help="Chips the chipper")(self.chip)
+        self.app.command(short_help="Kicks the kicker")(self.kick)
+        self.app.command(short_help="Show Robot Status Info")(self.stats)
+        self.app.command(short_help="Shows Redis Values")(self.redis)
+        self.app.command(short_help="Prints Thunderloop Logs")(self.log)
+        self.app.command(short_help="Prints Thunderloop Status")(self.status)
+        self.app.command(short_help="Restarts Thunderloop")(self.restart_thunderloop)
+        # Communication object responsible for proto execution/transmission
+        self.embedded_communication = embedded_communication
+        # Data handler responsible for disk information
+
+        self.embedded_data = self.embedded_communication.embedded_data
```

Comment:
Awkward extra line break
```suggestion
        # Communication object responsible for proto execution/transmission
        self.embedded_communication = embedded_communication
        # Data handler responsible for disk information
        self.embedded_data = self.embedded_communication.embedded_data
```